test/integration/infrastructure/infrastructure_test.go

```diff
@@ -0,0 +1,614 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var (
+	serviceAccount = flag.String("service-account", "", "Service account containing credentials for the GCP API")
+	region         = flag.String("region", "", "GCP region")
+)
+
+func validateFlags() {
+	if len(*serviceAccount) == 0 {
+		panic("--service-account flag is not specified")
+	}
+	if len(*region) == 0 {
+		panic("--region flag is not specified")
+	}
+}
+
+var _ = Describe("Infrastructure tests", func() {
+
+	var (
+		ctx    = context.Background()
+		logger *logrus.Entry
+
+		testEnv   *envtest.Environment
+		mgrCancel context.CancelFunc
+		c         client.Client
+
+		project        string
+		computeService *compute.Service
+		iamService     *iam.Service
+
+		internalChartsPath string
+	)
+
+	BeforeSuite(func() {
+		flag.Parse()
+		validateFlags()
+
+		internalChartsPath = gcp.InternalChartsPath
+		repoRoot := filepath.Join("..", "..", "..")
+		gcp.InternalChartsPath = filepath.Join(repoRoot, gcp.InternalChartsPath)
+
+		// enable manager logs
+		logf.SetLogger(zap.LoggerTo(GinkgoWriter, true))
+
+		log := logrus.New()
+		log.SetOutput(GinkgoWriter)
+		logger = logrus.NewEntry(log)
+
+		By("starting test environment")
+		testEnv = &envtest.Environment{
+			UseExistingCluster: pointer.BoolPtr(true),
+			CRDInstallOptions: envtest.CRDInstallOptions{
+				Paths: []string{
+					filepath.Join(repoRoot, "example", "20-crd-cluster.yaml"),
+					filepath.Join(repoRoot, "example", "20-crd-infrastructure.yaml"),
+				},
+			},
+		}
+
+		cfg, err := testEnv.Start()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(cfg).NotTo(BeNil())
+
+		By("setup manager")
+		mgr, err := manager.New(cfg, manager.Options{})
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(extensionsv1alpha1.AddToScheme(mgr.GetScheme())).To(Succeed())
+		Expect(gcpinstall.AddToScheme(mgr.GetScheme())).To(Succeed())
+
+		Expect(infrastructure.AddToManager(mgr)).To(Succeed())
+
+		var mgrContext context.Context
+		mgrContext, mgrCancel = context.WithCancel(ctx)
+
+		By("start manager")
+		go func() {
+			err := mgr.Start(mgrContext.Done())
+			Expect(err).NotTo(HaveOccurred())
+		}()
+
+		c = mgr.GetClient()
+		Expect(c).NotTo(BeNil())
+
+		project, err = gcp.ExtractServiceAccountProjectID([]byte(*serviceAccount))
+		Expect(err).NotTo(HaveOccurred())
+		computeService, err = compute.NewService(ctx, option.WithCredentialsJSON([]byte(*serviceAccount)), option.WithScopes(compute.CloudPlatformScope))
+		Expect(err).NotTo(HaveOccurred())
+		iamService, err = iam.NewService(ctx, option.WithCredentialsJSON([]byte(*serviceAccount)))
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterSuite(func() {
+		defer func() {
+			By("stopping manager")
+			mgrCancel()
+		}()
+
+		By("running cleanup actions")
+		framework.RunCleanupActions()
```

Comment:
Can you put this into `AfterEach`? ATM the `delete infrastructure` step is executed after both cases (non-existing/existing vpc) have finished, which I find to be a bit confusing.
I would expect, the cases to delete the infra directly after the creation was verified. WDYT?